Sources/ServiceLifecycle/GracefulShutdown.swift

```diff
@@ -80,12 +80,12 @@ public func cancelOnGracefulShutdown<T>(_ operation: @Sendable @escaping () asyn
         }
 
         let result = try await group.next()
+        group.cancelAll() // currently needed, as the group does not always auto-cancel tasks on return
```

Comment:
Hmm the comment seems a bit off, this is not accidental the group just waits, there's no implicit cancel at end of it.
\
Code looks good, just rephrase the comment?

cc @FranzBusch

Reply:
oh my, totally my fault, thanks for addressing this! I was already feeling bad about spreading misinformation in the github issue.

I was confusing the `cancelAll` behavior with [this](https://forums.swift.org/t/swift-5-8-throwingtaskgroup-waitforall/63519/4) for whatever reason (pretty wild in hindsight how I could mix these up). ...oh the shame... ; )

Reply:
may I offer a PR for the fix:
https://github.com/swift-server/swift-service-lifecycle/pull/137